src/platform/web.rs

```diff
@@ -436,3 +503,192 @@ impl Display for CustomCursorError {
+            ),
+            MonitorPermissionError::Prompt => write!(
+                f,
+                "User has not decided to give permission to query detailed monitor information"
+            ),
+            MonitorPermissionError::Unsupported => {
+                write!(f, "Browser does not support detailed monitor information")
+            },
+        }
+    }
+}
+
+impl Error for MonitorPermissionError {}
+
+#[cfg(not(web_platform))]
+struct PlatformHasMonitorPermissionFuture;
+
+#[derive(Debug)]
+pub struct HasMonitorPermissionFuture(PlatformHasMonitorPermissionFuture);
+
+impl Future for HasMonitorPermissionFuture {
+    type Output = bool;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        Pin::new(&mut self.0).poll(cx)
+    }
+}
+
+/// Additional methods on [`MonitorHandle`] that are specific to the Web.
+pub trait MonitorHandleExtWeb {
+    /// Returns whether the screen is internal to the device or external.
+    ///
+    /// External devices are generally manufactured separately from the device they are attached to
+    /// and can be connected and disconnected as needed, whereas internal screens are part of
+    /// the device and not intended to be disconnected.
+    fn is_internal(&self) -> Option<bool>;
+
+    /// Returns screen orientation data for this monitor.
+    fn orientation(&self) -> OrientationData;
+
+    /// Lock the screen orientation. The returned [`OrientationLockFuture`] can be dropped without
+    /// aborting the request.
+    ///
+    /// Will fail if another locking call is in progress.
+    fn request_lock(&self, orientation: OrientationLock) -> OrientationLockFuture;
+
+    /// Unlock the screen orientation.
+    ///
+    /// Will fail if a locking call is in progress.
+    fn unlock(&self) -> Result<(), OrientationLockError>;
+
+    /// Returns whether this [`MonitorHandle`] was created using detailed monitor permissions.
+    ///
+    /// See [`ActiveEventLoop::request_detailed_monitor_permission()`].
+    fn is_detailed(&self) -> bool;
+}
+
+impl MonitorHandleExtWeb for MonitorHandle {
+    fn is_internal(&self) -> Option<bool> {
+        self.inner.is_internal()
+    }
+
+    fn orientation(&self) -> OrientationData {
+        self.inner.orientation()
+    }
+
+    fn request_lock(&self, orientation_lock: OrientationLock) -> OrientationLockFuture {
+        OrientationLockFuture(self.inner.request_lock(orientation_lock))
+    }
+
+    fn unlock(&self) -> Result<(), OrientationLockError> {
+        self.inner.unlock()
+    }
+
+    fn is_detailed(&self) -> bool {
+        self.inner.is_detailed()
+    }
+}
+
+/// Screen orientation data.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct OrientationData {
+    /// The orientation.
+    pub orientation: Orientation,
+    /// [`true`] if the [`orientation`](Self::orientation) is flipped upside down.
+    pub flipped: bool,
```

Comment:
I'd just merge the `flipped` into `Orientation`

Reply:
This would require four variants, e.g.
```
enum Orientation {
    Landscape,
    LandscapeFlipped,
    Portrait,
    PortraitFlipped,
}
```
Adding natural would add two more, so six, all with very similar names.
I think this makes it harder to extract information and be more concise about each field and name.

So I consider the current implementation better.

Reply:
I was thinking to probably make the type more general in this stage, given that on Wayland I have the following `https://wayland.app/protocols/wayland#wl_output:enum:transform`

So I think `flipped` could also have different meaning...

I'd assume on Web it's just rotated and not really flipped?

Reply:
No rotating would be between "rotating between landscape and potrait".
Flipped means vertical axis indeed, but that can apply to both landscape and portrait.

I don't know how to interpret the Wayland data, because it doesn't actually contain `Orientation`, it only contains if its flipped and in which degree, but not if its landscape or portrait.
I guess this can be inferred by comparing the width and height?

Reply:
We figured out that Wayland transform is probably not the same as screen orientation.